src/components/datagrid/datagrid.tsx

```diff
@@ -0,0 +1,227 @@
+    const rowUrl = urlField ? rowData[urlField] : null;
+    const data = rowData[field];
+    const type = typeof data;
+
+    // Explicit link: passed as URL without being set as urlField.
+    const isExplicitLink = String(data).match(REGEX_URL);
+
+    // Implicit link: first column and `rowUrl` resolved using `urlFields`.
+    const isImplicitLink = rowUrl && fieldIndex === 0;
+
+    // Cell should be a link is truthy.
+    const link = isExplicitLink
+      ? String(data)
+      : isImplicitLink
+        ? String(rowUrl)
+        : "";
+
+    let contents: React.ReactNode = data;
+    switch (type) {
+      case "boolean":
+        contents = (
+          <Boolean {...(booleanProps as BooleanProps)} value={!!data} />
+        );
+        break;
+      case "number":
+        contents = <Badge {...(badgeProps as BadgeProps)}>{data}</Badge>;
+    }
+
+    return (
+      <td
+        className={clsx(
+          "mykn-datagrid__cell",
+          `mykn-datagrid__cell--type-${type}`,
+        )}
+        aria-description={field2Caption(field as string)}
+        key={`${id}-row-${index}-${String(field)}`}
+      >
+        {isExplicitLink ? (
+          <A href={link} target="_blank" {...aProps}>
+            {contents}
+          </A>
+        ) : (
+          <>
+            {isImplicitLink ? (
+              <P>
+                <A href={link} aria-label={link}>
+                  <Outline.ArrowTopRightOnSquareIcon />
+                </A>
+                &nbsp;
+                {contents}
+              </P>
+            ) : (
+              <P>{contents}</P>
+            )}
+          </>
+        )}
+      </td>
+    );
+  };
+
+  return (
+    <div className="mykn-datagrid" {...props}>
+      {/* Caption */}
+      <table
+        className="mykn-datagrid__table"
+        role="grid"
+        aria-labelledby={titleId}
+      >
+        {title && (
+          <caption className="mykn-datagrid__caption">
+            <H3 id={titleId as string}>{title}</H3>
+          </caption>
+        )}
+
+        {/* Headings */}
+        <thead className="mykn-datagrid__head" role="rowgroup">
+          <tr className="mykn-datagrid__row" role="row">
+            {captions.map((caption) => (
+              <th
+                key={`${id}-heading-${String(caption)}`}
+                className="mykn-datagrid__cell mykn-datagrid__cell--header"
+                role="columnheader"
+              >
+                <P bold muted size="xs">
+                  {caption}
+                </P>
+              </th>
+            ))}
+          </tr>
+        </thead>
+
+        {/* Cells */}
+        <tbody className="mykn-datagrid__body" role="rowgroup">
+          {results.map((rowData, index) => (
+            /**
+             * FIXME: This effectively still uses index as keys which might lead to issues.
+             * @see {@link https://react.dev/learn/rendering-lists#rules-of-keys|Rules of keys}
+             */
```

Comment:
Could we apply the FIXME? :smile:

Reply:
Will do so in a future PR containing the sorting functionality.